### PR TITLE
Introduce ForbiddenClassNameExtension for append additional forbidden…

### PR DESCRIPTION
**Pull request by kamil-zacek**
… class prefixes

### CODE REVIEW

**Comment by staabm** [comment]
Could you give us an idea what your use-case is?

Is it php-scoper related or something completely different?

**Comment by ondrejmirtes** [comment]
@staabm This was discussed outside of GitHub. The idea is that you want to report usages of classes that you're not supposed to use in application. For example: Generated Doctrine proxies from their configured namespace.

File: src/Classes/ForbiddenClassNameExtension.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Classes;
+
+interface ForbiddenClassNameExtension
```
Comment (ondrejmirtes) [inline src/Classes/ForbiddenClassNameExtension.php:24]: Please put `@api` here and explanation what it is for and how to register it. Example: https://apiref.phpstan.org/1.11.x/source-src.PhpDoc.StubFilesExtension.html#5

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
The only other problem is that this isn't tested. Please create a new RuleTestCase for one of the rules affected by this, add an example new neon file with a new extension implementing this interface, and reference it in `public static function getAdditionalConfigFiles()` in that test case. That way this extension will be added to the container.

**Review by ondrejmirtes — APPROVED** [review]

**Comment by ondrejmirtes** [comment]
Perfect, thank you!

**Pull request merged**